hoprd/rest-api/src/session.rs

```diff
@@ -22,24 +24,42 @@ use crate::{ApiErrorStatus, InternalState, BASE_PATH};
         "path": {
             "Hops": 1
         },
-        "port": 0
+        "protocol": "TCP",
+        "target": "localhost:8080",
+        "listen_port": 10000,
+        "capabilities": ["Retransmission", "Segmentation"]
     }))]
 #[serde(rename_all = "camelCase")]
 pub(crate) struct SessionClientRequest {
     #[serde_as(as = "DisplayFromStr")]
     pub destination: PeerId,
-    pub path: PathOptions,
-    #[serde(default)]
-    pub port: u16,
+    pub path: RoutingOptions,
+    pub protocol: IpProtocol,
+    pub target: String,
+    pub listen_port: Option<u16>,
+    #[serde_as(as = "Option<Vec<DisplayFromStr>>")]
+    pub capabilities: Option<Vec<SessionCapability>>,
```

Comment:
Nope, because there's a different default for TCP (`[Retransmission,Segmentation]`) and different one for UDP (`[]`). So if I want TCP and don't specify anything, it will default to `Retransmission` and `Segmentation`, whereas if I specify `[]` with TCP, I can override it.

Reply:
Would you ever want to do that though>?

Reply:
If something does not work, it might be useful to have the ability to override to debug stuff